gems/aws-sdk-core/lib/aws-sdk-core/plugins/retries/error_inspector.rb

```diff
@@ -106,14 +118,20 @@ def modeled_throttling?
           @error.is_a?(Errors::ServiceError) && @error.throttling?
         end
 
+        def clock_skew?(context)
+          CLOCK_SKEW_ERRORS.include?(@name) &&
+            context.config.clock_skew.clock_skewed?(context)
+        end
+
         def retryable?(context)
           (expired_credentials? && refreshable_credentials?(context)) ||
             throttling_error? ||
             checksum? ||
             networking? ||
             server? ||
             endpoint_discovery?(context) ||
-            modeled_retryable?
+            modeled_retryable? ||
+            clock_skew?(context)
```

Comment:
Does any ordering matter here? I would think that this should be after `server?` -- also maybe `modeled_retryable` should be moved up.

Reply:
In theory order should not matter here - I guess in theory we want to put the most likely errors first (to take advantage of short circuiting).  However, because the check for endpoint discovery actually will do something, the order *could* matter in some cases (see other discussion).  I've updated the order a bit.